plugin/tools.py

```diff
@@ -702,34 +702,68 @@ def get_clang_version_str(clang_binary):
         log.info("Getting version from command: `%s`", check_version_cmd)
         output_text = Tools.run_command(check_version_cmd, shell=True)
 
+        if sublime.platform() == "osx":
+            return cls._get_clang_version_str_osx(output_text)
+        elif sublime.platform() == "linux":
+            return cls._get_clang_version_str_linux(output_text)
+        elif sublime.platform() == "windows":
+            return cls._get_clang_version_windows(output_text)
+        else:
+            log.error("Unknown platform: %s", sublime.platform())
+            return None
+
+    @classmethod
+    def _get_clang_version_str_osx(cls, output_text):
+        # There can be two flavours on osx: the apple flavour or the trunk/brew
+        # flavour.
+        if "Apple" in output_text:
+            return cls._get_apple_clang_version_str(output_text)
+        else:
+            # Same as on linux
+            return cls._get_clang_version_str_linux(output_text)
```

Comment:
I still don't really like having the name of the os in the function name. Basically, we have 2 ways to get the version: `clang` and `apple_clang`. We should have 2 functions: `_get_apple_clang_version_str` and `_get_clang_version_str`. Then all the boilerplate functions become unnecessary. These functions should just be called inline.